lightningd/gossip/gossip.c

```diff
@@ -418,15 +418,27 @@ static struct io_plan *release_peer(struct io_conn *conn, struct daemon *daemon,
 			      "%s", tal_hex(trc, msg));
 
 	peer = find_peer(daemon, unique_id);
-	if (!peer)
-		status_failed(WIRE_GOSSIPSTATUS_BAD_RELEASE_REQUEST,
-			      "Unknown peer %"PRIu64, unique_id);
+	if (!peer) {
+		/* This can happen with a reconnect vs connect race.
+		 * See gossip_peer_released in master daemon. */
+		struct crypto_state dummy;
```

Comment:
Yup, this is not as nice as it could be, we could introduce explicit error messages that avoid having to send dummy messages. Setting the MSB or `~type` might be a simple mechanism to make use of the existing case distinction.